homeassistant/components/generic/camera.py

```diff
@@ -125,6 +125,11 @@ def camera_image(self):
         ).result()
 
     async def async_camera_image(self):
+        """Wrap _async_camera_image with an asyncio.shield."""
+        # Shield the request because of https://github.com/encode/httpx/issues/1461
+        return await asyncio.shield(self._async_camera_image())
+
+    async def _async_camera_image(self):
         """Return a still image response from the camera."""
```

Comment:
I think it was better when you only shielded the httpx call. Anything inside the shield will continue to execute, but the surrounding code is cancelled. That mean that the setting of last image/url will still end up being set sometime later even though the command was cancelled.

So just shield the "buggy" code and let the rest be cancelled as expected.

Reply:
That way, we avoid setting last url/image if the httpx stuff ends up completing even though caller wanted it cancelled.

Reply:
The reason I was concerned about doing it that way was in the event of a cancellation event, the call would continue inside the shield but the `response` resources might not get freed up afterwards since the `response` is never read.
If we don't want to set the last_url/image I can still change that. I do think that does make more sense technically (the caller requested a cancel, so cancel), but practically it might make more sense to just update the last_url/image since we have it anyway.

Reply:
@elupus Any thoughts on the above?

Reply:
Well the cancel can occur on any await. So there is a risk of introducing a later await that changes behavior.

Being as specific as possible what it is we are solving is better. If cancelled we should throw away anything we get..a new request may have been started.

Reply:
Ok, the change that was pushed last week does avoid setting the last_url/image. I agree we should only shield one await, but we also need to make sure the `response` gets closed, and this way satisfies both of those requirements.

Reply:
@elupus Does what I said make sense? If we only shield the call then we risk not cleaning up the response.
It looks like another user had the same problem, so it would be good to merge a fix before another release.

Reply:
Yes i understand.